### ISSUE TITLE
Cannot Instantiate PFUser Query

### ISSUE THREAD

**Issue by natecraft1** [ParsePlatform/Parse-SDK-iOS-OSX#176]
After updating parse, my code gets hung up on this line:
let userQuery = PFUser.query()!

It does not throw any error.
I'm guessing the class is missing something like:

+ (PFQuery *)query {
    return [super query];
}

I don't really know, but I didn't see anything related to Query in the PFUser class.

**Comment by nlutsenko** [comment]
Hey @natecraft1, thanks for the report.
`+ (PFQuery *)query` is actually implemented inside the PFObject implemtation.
I've just tested this and current master works fabulously whenever this method is called.

Any chance you can pause the application and post a stack trace of the currently locked thread here?

**Comment by natecraft1** [comment]
@nlutsenko Thanks for getting back.  How do I post a stack trace?  
I put a breakpoint at these 3 lines:
       let query: PFQuery = PFInstallation.query()!
        let userQuery = PFUser.query()!
        print("doesnt get here")

I stepover the breakpoint on the first line and it shows a lot of stuff on the main thread.  I stepover the second line and my app resumes playing and I see no thread information, and line 3 is never reached.  Thanks so much

**Comment by nlutsenko** [comment]
This is super strange, since I can't reproduce this at all.
I've just used a starter project that is contained in this repository (use it from releases or from source).
Replaced `ViewController.swift` with this code:
```swift
import UIKit
import Parse

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        Parse.setApplicationId("asdf", clientKey: "asdf")
        let installationQuery = PFInstallation.query()!
        let userQuery = PFUser.query()!
        print(installationQuery)
        print(userQuery)
        print("yolo")
    }
}
```
And can see the following logged to console: `<PFQuery: 0x7fd7ab225760><PFQuery: 0x7fd7a968bb60>yolo`
Any chance you can try it out on a test project? Also - please make sure you set the app id/client key before asking for a query, since it's not going to work without it.

**Comment by natecraft1** [comment]
@nlutsenko  I think I figured it out.  It happens because I am subclassing user.  Add this code to your test project and then print(PFUser.query())

class User: PFUser {
    
    override class func initialize() {
        struct Static {
            static var onceToken : dispatch_once_t = 0;
        }
        dispatch_once(&Static.onceToken) {
            self.registerSubclass()
        }
    }
}

**Comment by nlutsenko** [comment]
Just tested with the following:
```swift
import UIKit
import Parse

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        Parse.setApplicationId("asdf", clientKey: "asdf")
        let installationQuery = PFInstallation.query()!
        let userQuery = PFUser.query()!
        let customUserQuery = User.query()!
        print(installationQuery, installationQuery.parseClassName)
        print(userQuery, userQuery.parseClassName)
        print(customUserQuery, customUserQuery.parseClassName)
        print("yolo")
    }
}

class User: PFUser {
    override class func initialize() {
        struct Static {
            static var onceToken : dispatch_once_t = 0;
        }
        dispatch_once(&Static.onceToken) {
            self.registerSubclass()
        }
    }
}
```
It succesfully logs:
```
<PFQuery: 0x7f9f45a022e0> _Installation
<PFQuery: 0x7f9f43d34a00> _User
<PFQuery: 0x7f9f43d352c0> _User
yolo
```

**Comment by natecraft1** [comment]
@nlutsenko Oh, sorry.  I think you have to instantiate a User.  Try putting    
let user = User()
at the top of the ViewController

**Comment by nlutsenko** [comment]
I think I found a source of problem! Woohoo!
Going to send a Pull Request in a moment, will ping you after so you can test from a branch (as well as just apply the changes manually and see if it works)

**Comment by natecraft1** [comment]
@nlutsenko Awesome thanks so much!

**Comment by nlutsenko** [comment]
Ok, @natecraft1 [this](https://github.com/ParsePlatform/Parse-SDK-iOS-OSX/pull/180) pull request hopefully helps. It's basically 1 line of code - inside PFObject.m
Try patching it locally and let me know if it works.

**Comment by nlutsenko** [comment]
Some notes: on a branch of that Pull Request - everything works great, with instances of PFUser() and User() as well as both queries.

**Comment by natecraft1** [comment]
@nlutsenko Hey, I see you have 3 diffs.  I just changed isKindOfClass to isSubclassOfClass locally and it seems to work!  Now we can yolo = ).  Is there anything else I need to do?  And do I just leave the fix locally and wait for a parse update?

**Comment by nlutsenko** [comment]
Yolo! =) (yup, did a little bit more stuff while investigating your problem)
You can leave the fix locally, it will be included in the next version (1.8.3)
Have a great rest of your Sunday!
